spec/mailers/publish_mailer_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+RSpec.describe PublishMailer do
+  let(:recipient) { build(:user) }
+
+  describe "#publish_email" do
+    it "creates an email for the user" do
+      edition = build(:edition, :published, base_path: "/news/some-great-news")
+
+      mail = PublishMailer.publish_email(recipient, edition, edition.status)
+      mail_subject = I18n.t!("publish_mailer.publish_email.subject.published",
+                             title: edition.title)
+
+      expect(mail.to).to eq([recipient.email])
+      expect(mail.subject).to eq(mail_subject)
+      expect(mail.body).to include("https://www.test.gov.uk/news/some-great-news")
+      expect(mail.body).to include(document_path(edition.document))
+    end
+
+    it "includes when the edition was published" do
+      edition = build(:edition,
+                      :published,
+                      published_at: Time.zone.parse("2019-06-17 9:00"))
+
+      mail = PublishMailer.publish_email(recipient, edition, edition.status)
+      publish_time = I18n.t!("publish_mailer.publish_email.details.publish",
+                             date: "17 June 2019",
+                             time: "9:00am",
+                             user: recipient.name)
+
+      expect(mail.body.to_s).to include(publish_time)
+    end
+
+    context "when the edition published needs 2i review" do
+      it "informs recipients" do
+        edition = build(:edition, :published, state: :published_but_needs_2i)
+
+        mail = PublishMailer.publish_email(recipient, edition, edition.status)
+
+        review_notice = I18n.t!("publish_mailer.publish_email.2i_warning")
+        expect(mail.body.to_s).to include(review_notice)
+      end
+    end
+
+    context "when a subsequent edition is published" do
```

Comment:
I think we should also check the subject here so that i18n key is executed. Might be able to include as part of one of the existing tests with a little rewording.

Reply:
Yep you're correct, I've added it to the test `"includes when the edition was updated"` as naming wise it makes sense for it to be there.